software/contrib/master_clock.py

```diff
@@ -241,24 +218,26 @@ def screen2(self):
                 # unlock the knob if it has reached near the same value - avoids messy UX
                 if abs(newBpm - self.state.get('bpm')) <= 10:
                     self.k2Unlocked = True
-                # update config value if k2 is unlocked
-                if self.k2Unlocked:
+                # update config value if k2 is unlocked and bpm has changed
+                if self.k2Unlocked: # and self.bpm != newBpm:
```

Comment:
I'm guessing the second half of this line was commented out for testing? Should it be added back into the conditional check?

Reply:
It was actually this update that caused the bug that prevented the Pulse length from being editable. This is the because the pulse length can be be updated separately from bpm. I will remove that part now from my repo so it is reflected during the merge.